src/teleop/gui/src/components/SATask.vue

```diff
@@ -1,20 +1,314 @@
+      <PDBFuse/>
+    </div>
+    <div class="box light-bg jointState">
+      <JointStateTable v-bind:jointStateData="jointState" v-bind:vertical="true"/>
+    </div>
+    <div class="box light-bg moteus">
+      <MoteusStateTable v-bind:moteusStateData="moteusState"/>
+    </div>
+  </div>
 </template>
 
 <script>
+  import ROSLIB from "roslib"
+  import ERDMap from './ERDRoverMap.vue';
+  import ERDWaypointEditor from './ERDWaypointEditor.vue'
+  import DriveControls from "./DriveControls.vue";
+  import EndEffectorUV from "./EndEffectorUV.vue"
+  import ArmControls from "./ArmControls.vue"
+  import PDBFuse from "./PDBFuse.vue"
+  import * as qte from "quaternion-to-euler";
+  import Cameras from "./Cameras.vue";
+  import MoteusStateTable from './MoteusStateTable.vue';
+  import JointStateTable from './JointStateTable.vue';
 
-export default {
-  data() {
-    return {
 
+  export default {
+    data() {
+      return {
+        // Default coordinates are at NC 53 Parking Lot
+        odom: {
+          latitude_deg: 42.294864932393835,
+          longitude_deg: -83.70781314674628,
+          bearing_deg: 0
+        },
+
+        primary: true,
+
+        jointState: {},
+        // Default object isn't empty, so has to be initialized to ""
+        moteusState: {
+          name: ["", "", "", "", "", ""],
+          error: ["", "", "", "", "", ""],
+          state: ["", "", "", "", "", ""]
+        },
+
+        // Pubs and Subs
+        odom_sub: null,
+        tfClient: null
+      }
+    },
+
+    created: function () {
+      this.odom_sub = new ROSLIB.Topic({
+        ros: this.$ros,
+        name: '/gps/fix',
+        messageType: 'sensor_msgs/NavSatFix'
+      });
+
+      this.tfClient = new ROSLIB.TFClient({
+        ros: this.$ros,
+        fixedFrame: 'odom',
+        // Thresholds to trigger subscription callback
+        angularThres: 0.01,
+        transThres: 0.01
+      });
+
+      // Subscriber for odom to base_link transform
+      this.tfClient.subscribe('base_link', (tf) => {
+        // Callback for IMU quaternion that describes bearing
+        let quaternion = tf.rotation
+        quaternion = [quaternion.w, quaternion.x, quaternion.y, quaternion.z]
+        //Quaternion to euler angles
+        let euler = qte(quaternion)
+        // euler[2] == euler z component
+        this.odom.bearing_deg = euler[2] * (180 / Math.PI)
+        console.log(tf)
+      });
+
+      this.odom_sub.subscribe((msg) => {
+        // Callback for latLng to be set
+        this.odom.latitude_deg = msg.latitude
+        this.odom.longitude_deg = msg.longitude
+      });
+
+      this.brushless_motors = new ROSLIB.Topic({
+          ros: this.$ros,
+          name: 'drive_status',
+          messageType: 'mrover/MotorsStatus'
+      });
+
+      this.brushless_motors.subscribe((msg) => {
+          this.jointState = msg.joint_states
+          this.moteusState = msg.moteus_states
+      });
+    },
+
+    components:{
+      ERDMap,
```

Comment:
Let's rename the ERD stuff here, maybe just name it "BasicRoverMap" or something since it gets used on 3 different GUIs now. Same with ERDWaypointEditor.